src/pointer-analysis/value_set.cpp

```diff
@@ -1694,3 +1704,6 @@ exprt value_sett::make_member(
 
   return member_expr;
 }
+
+value_sett::expr_simplifiert value_sett::default_simplifier=
+  [](exprt &e, const namespacet &ns) { simplify(e, ns); };
```

Comment:
See above: I have no idea what this is good for.

Reply:
Also not sure. @smowton, do you have a use case for a custom simplifier?

Reply:
Yes, to prevent merging of identical structs (e.g. two Java objects with the same field names but unrelated types having casts erased)

Reply:
That's removing the symptom rather than the root cause, I'd say. If Java objects need to be handled differently (which part of the code base requires this?), then the Java front-end needs to make sure this happens. We can't go across all the code base and build in special cases in each place.